src/tools/markup.ts

```diff
@@ -0,0 +1,299 @@
+  while (emptyDirPaths.length > 0) {
+    emptyDirPaths.forEach((path) => {
+      if (!dryRun) filesystem.remove(path)
+      removedDirs.push(path)
+    })
+    emptyDirPaths = getEmptyDirPaths()
+  }
+
+  return removedDirs
+}
+
+export async function deleteFiles({
+  filePaths,
+  comment,
+  dryRun = true,
+}: {
+  filePaths: string[]
+  comment: string
+  dryRun?: boolean
+}) {
+  const commentResults = await Promise.allSettled(
+    filePaths.map(async (path) => {
+      const { remove } = filesystem
+      const { exists } = patching
+      if (await exists(path, comment)) {
+        if (!dryRun) {
+          remove(path)
+        }
+        return { path }
+      }
+    }),
+  )
+
+  return commentResults
+}
+
+// remove file action
+// all other actions update the contents of the file
+
+export async function updateFiles({
+  filePaths,
+  markupPrefix,
+  dryRun = true,
+  removeMarkupOnly = false,
+}: {
+  filePaths: string[]
+  markupPrefix: string
+  dryRun?: boolean
+  removeMarkupOnly?: boolean
+}) {
+  const sanitize = (contents: string) => {
+    return contents.replace(markupRegex(markupPrefix), "")
+  }
+
+  // Go through every file path and handle the operation for each comment
+  const commentResults = await Promise.allSettled(
+    filePaths.map(async (path) => {
+      const { exists, update } = patching
+      const { read } = filesystem
+
+      const comments: string[] = []
+
+      // remove files first
+      if (await exists(path, markupComment(markupPrefix, MarkupComments.RemoveFile))) {
+        console.log("removing file", path)
+        if (!dryRun) {
+          if (removeMarkupOnly) {
+            const contents = read(path)
+            const sanitized = sanitize(contents)
+            filesystem.write(path, sanitized)
+          } else {
+            filesystem.remove(path)
+          }
+        }
+        comments.push(MarkupComments.RemoveFile)
+        return { path, comments }
+      }
+
+      // filter out RemoveFile (weve already handled it above)
+      // and create a regex for the remaining comment types
+      const operationComments = Object.keys(MarkupComments)
+        .filter((key) => MarkupComments[key] !== MarkupComments.RemoveFile)
+        .map((key) => markupComment(markupPrefix, MarkupComments[key]))
+
+      const shouldUpdate = removeMarkupOnly
+        ? markupRegex(markupPrefix)
+        : RegExp(operationComments.join("|"), "g")
+
+      if (await exists(path, shouldUpdate)) {
+        const before = read(path)
+
+        operationComments.forEach((operation) => {
+          if (before.includes(operation)) {
+            comments.push(operation)
+          }
+        })
+
+        if (!dryRun) {
+          await update(path, (contents: string) => {
+            let result = contents
+            if (removeMarkupOnly) {
+              result = sanitize(result)
+            } else {
+              result = updateFile(result, markupPrefix)
+              result = sanitize(result)
```

Comment:
I don't know if this extra `sanitize()` was happening before - what behavior do we want here, I assumed we'd always want to get rid of all the markup regardless